Mathlib/CategoryTheory/Shift/SingleFunctors.lean

```diff
@@ -0,0 +1,220 @@
+  hom (a : A) : F.functor a ⟶ G.functor a
+  comm (n a a' : A) (ha' : n + a = a') : (F.shiftIso n a a' ha').hom ≫ hom a =
+    whiskerRight (hom a') (shiftFunctor D n) ≫ (G.shiftIso n a a' ha').hom := by aesop_cat
+
+namespace Hom
+
+attribute [reassoc] comm
+attribute [local simp] comm comm_assoc
+
+/-- The identity morphism in `SingleFunctors C D A`. -/
+@[simps]
+def id : Hom F F where
+  hom a := 𝟙 _
+
+variable {F G H}
+
+/-- The composition of morphisms in `SingleFunctors C D A`. -/
+@[simps]
+def comp (α : Hom F G) (β : Hom G H) : Hom F H where
+  hom a := α.hom a ≫ β.hom a
+
+end Hom
+
+instance : Category (SingleFunctors C D A) where
+  Hom := Hom
+  id := Hom.id
+  comp := Hom.comp
+
+@[simp]
+lemma id_hom (a : A) : Hom.hom (𝟙 F) a = 𝟙 _ := rfl
+
+variable {F G H}
+
+@[simp, reassoc]
+lemma comp_hom (f : F ⟶ G) (g : G ⟶ H) (a : A) : (f ≫ g).hom a = f.hom a ≫ g.hom a := rfl
+
+@[ext]
+lemma hom_ext (f g : F ⟶ G) (h : f.hom = g.hom) : f = g := Hom.ext f g h
+
+/-- Constructor for isomorphisms in `SingleFunctors C D A`. -/
+@[simps]
+def isoMk (iso : ∀ a, (F.functor a ≅ G.functor a))
+    (comm : ∀ (n a a' : A) (ha' : n + a = a'), (F.shiftIso n a a' ha').hom ≫ (iso a).hom =
+      whiskerRight (iso a').hom (shiftFunctor D n) ≫ (G.shiftIso n a a' ha').hom) :
+    F ≅ G where
+  hom :=
+    { hom := fun a => (iso a).hom
+      comm := comm }
+  inv :=
+    { hom := fun a => (iso a).inv
+      comm := fun n a a' ha' => by
+        dsimp only
+        rw [← cancel_mono (iso a).hom, assoc, assoc, Iso.inv_hom_id, comp_id, comm,
+          ← whiskerRight_comp_assoc, Iso.inv_hom_id, whiskerRight_id', id_comp] }
+
+variable (C D)
+
+/-- The evaluation `SingleFunctors C D A ⥤ C ⥤ D` for some `a : A`. -/
+@[simps]
+def evaluation (a : A) : SingleFunctors C D A ⥤ C ⥤ D where
+  obj F := F.functor a
+  map {F G} φ := φ.hom a
+
+variable {C D}
+
+@[reassoc (attr := simp)]
+lemma hom_inv_id_hom (e : F ≅ G) (n : A) : e.hom.hom n ≫ e.inv.hom n = 𝟙 _ := by
+  rw [← comp_hom, e.hom_inv_id, id_hom]
+
+@[reassoc (attr := simp)]
+lemma inv_hom_id_hom (e : F ≅ G) (n : A) : e.inv.hom n ≫ e.hom.hom n = 𝟙 _ := by
+  rw [← comp_hom, e.inv_hom_id, id_hom]
+
+@[reassoc (attr := simp)]
+lemma hom_inv_id_hom_app (e : F ≅ G) (n : A) (X : C) :
+    (e.hom.hom n).app X ≫ (e.inv.hom n).app X = 𝟙 _ := by
+  rw [← NatTrans.comp_app, hom_inv_id_hom, NatTrans.id_app]
+
+@[reassoc (attr := simp)]
+lemma inv_hom_id_hom_app (e : F ≅ G) (n : A) (X : C) :
+    (e.inv.hom n).app X ≫ (e.hom.hom n).app X = 𝟙 _ := by
+  rw [← NatTrans.comp_app, inv_hom_id_hom, NatTrans.id_app]
+
+instance (f : F ⟶ G) [IsIso f] (n : A) : IsIso (f.hom n) :=
+  (inferInstance : IsIso ((evaluation C D n).map f))
+
+variable (F)
+
+/-- The postcomposition `SingleFunctors C D A → SingleFunctors C E A` induced
+by a functor `D ⥤ E` which commutes with the shift by `A`. -/
+@[simps! functor shiftIso_hom_app shiftIso_inv_app]
+def postcomp (G : D ⥤ E) [G.CommShift A] :
+    SingleFunctors C E A where
```

Comment:
How is this used most often? Do you ever have a fixed `G` and consider the induced map  `SingleFunctors C D A → SingleFunctors C E A` (the doc string seems to suggest this)? If yes, the order of the arguments should maybe be switched such that `G` comes first?

Reply:
Thanks for the suggestions. I have changed the docstring.